### PR TITLE
Removing upgrade documentation

### PR DESCRIPTION
**Pull request by l0k0ms**
Removing upgrade documentation from the repo, and leaving a placeholder to point to the public documentation page instead.

### CODE REVIEW

File: docs/agent/upgrade.md [hunk 1]
```diff
@@ -1,12 +1,5 @@
 # Upgrade to Agent 6
 
-Datadog Agent 6 has officially been released. Please use the in-application [install instructions](https://app.datadoghq.com/account/settings#agent) to install Agent 6 from scratch or upgrade from Agent 5.
+Instructions on how to upgrade your Agent from version 5 to version 6 can be found in [Datadog public documentation](https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/).
```
Comment (l0k0ms) [inline docs/agent/upgrade.md]:
```suggestion
Instructions to upgrade your Agent from version 5 to version 6 can be found in [Datadog public documentation](https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/).
```

File: docs/agent/upgrade.md [hunk 2]
```diff
@@ -1,12 +1,5 @@
 # Upgrade to Agent 6
 
-Datadog Agent 6 has officially been released. Please use the in-application [install instructions](https://app.datadoghq.com/account/settings#agent) to install Agent 6 from scratch or upgrade from Agent 5.
+Instructions to upgrade your Agent from version 5 to version 6 can be found in [Datadog public documentation](https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/).
```
Comment (apigirl) [inline docs/agent/upgrade.md]:
```suggestion
Instructions to upgrade your Agent from version 5 to version 6 can be found in the [Datadog public documentation](https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/).
```

**Review by apigirl — DISMISSED** [review]

**Review by albertvaka — DISMISSED** [review]

**Review by apigirl — DISMISSED** [review]

File: docs/agent/upgrade.md [hunk 3]
```diff
@@ -1,12 +1,3 @@
...
-
-For a list of changes between Agent 5 and 6 that may affect you, please refer to:
-* [the doc on the general Agent changes][changes]
-* [the doc on the differences in the Agent config options][config]
-* [the list of Agent 5 features that are not ported to Agent 6][missing-features]
-
-[changes]: changes.md
-[config]: config.md
-[missing-features]: missing_features.md
+Instructions to upgrade your Agent from version 5 to version 6 can be found in the [Datadog public documentation](https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/).
```
Comment (hush-hush) [inline docs/agent/upgrade.md]: Should we use the `#upgrade-to-agent-6` anchor https://docs.datadoghq.com/agent/guide/upgrade-to-agent-v6/#upgrade-to-agent-6 ?

Comment (l0k0ms) [reply docs/agent/upgrade.md]: No need i think, If people do reach this page through the datadog-agent repo, having some deductive approach where Agent v6 is explained before instructions on how to install/upgrade it are given can't hurt.

Comment (hush-hush) [reply docs/agent/upgrade.md]: ok

**Review by hush-hush — APPROVED** [review]

**Pull request merged**